cmd/soci/commands/index/list.go

```diff
@@ -24,17 +24,38 @@ import (
 
 	"github.com/awslabs/soci-snapshotter/soci"
 	"github.com/containerd/containerd/cmd/ctr/commands"
+	"github.com/containerd/containerd/images"
 	"github.com/containerd/containerd/platforms"
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/urfave/cli"
 )
 
+type filter func(ae *soci.ArtifactEntry) bool
+
 func indexFilter(ae *soci.ArtifactEntry) bool {
 	return ae.Type == soci.ArtifactEntryTypeIndex
 }
 
-func originalDigestFilter(digest string) func(*soci.ArtifactEntry) bool {
+func platformFilter(platform specs.Platform) filter {
+	return func(ae *soci.ArtifactEntry) bool {
+		return indexFilter(ae) && ae.Platform == platforms.Format(platform)
+	}
+}
+
+func originalDigestFilter(digest string) filter {
+	return func(ae *soci.ArtifactEntry) bool {
+		return indexFilter(ae) && ae.OriginalDigest == digest
+	}
+}
+
+func any(fns []filter) filter {
```

Comment:
Maybe change this name? `any` isn't very descriptive. Also, conflicts with the `any` type.